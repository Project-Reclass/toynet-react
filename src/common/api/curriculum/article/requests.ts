/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import axios from 'axios';

export interface ArticleImage {
  file: string;
  credit: string;

  caption?: string;
}

export interface ArticleMeta {
  source: string;
  title: string;
  author: string;
  images: ArticleImage[];
}

export const getArticleText = async (articleId: number): Promise<string> => {
  const { data } = await axios.get(`/data/article/${articleId}/article.md`);
  return data;
};

export const getArticleMeta = async (articleId: number): Promise<ArticleMeta> => {
  const { data } = await axios.get(`/data/article/${articleId}/meta.json`);
  return data;
};