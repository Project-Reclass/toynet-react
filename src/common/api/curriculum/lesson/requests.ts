import axios from 'axios';

type LessonSlideURI = string;
export interface Meta {
  numSlides: number;
}

export const getLessonSlide = async (submoduleId: number, slideId: number): Promise<LessonSlideURI> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/${slideId}.png`);
  return data;
};

export const getLessonMeta = async (submoduleId: number): Promise<Meta | undefined> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/meta.json`);
  return data;
};
