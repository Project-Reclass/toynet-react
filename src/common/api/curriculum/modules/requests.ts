import { simulateNetwork } from 'src/common/utils';
import modules from 'src/__data__/curriculum/modules.json';

export const getAllModules = () => {
  return simulateNetwork(() => modules);
};

export const getModuleById = (id: number) => {
  return simulateNetwork(() => modules.find(module => module.id === id));
};

export const getModuleByName = (name: string) => {
  return simulateNetwork(() => modules.find(module => module.name === name));
};