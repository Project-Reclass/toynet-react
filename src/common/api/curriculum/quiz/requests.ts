import axios from 'axios';

export const getQuizMeta = async (quizId: number) => {
	const { data } = await axios.get(`/data/quiz/${quizId}/meta.json`);
	return data;
};