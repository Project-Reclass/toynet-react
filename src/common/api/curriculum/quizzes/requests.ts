import { simulateNetwork } from 'src/common/utils';
import quizzes from 'src/__data__/curriculum/quizzes.json';

export const getAllQuizzes = async () => {
  return simulateNetwork(() => quizzes);
};

export const getQuizById = (id: number) => {
  return simulateNetwork(() => (
    quizzes.find(quiz => quiz.id === id)
  ));
};
