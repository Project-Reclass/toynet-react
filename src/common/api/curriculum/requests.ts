import { simulateNetwork } from 'src/common/utils';
import curriculums from 'src/__data__/curriculum/curriculums.json';

export const getAllCurriculums = () => {
  return simulateNetwork(() => curriculums);
};

export const getCurriculumById = (id: number) => {
  return simulateNetwork(() => (
    curriculums.find(value => value.id === id)
  ));
};