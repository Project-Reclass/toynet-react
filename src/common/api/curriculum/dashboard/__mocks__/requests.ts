/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/

import { DashboardIntf } from 'src/common/types/curriculum';

const defaultMockData = {
  id: 1,
  name: 'test',
  introduction: 'intro',
  modules: [],
};

export let mockData: DashboardIntf = {...defaultMockData};

export function __resetMock() {
  mockData = {...defaultMockData};
}

export function __setMockData(data: DashboardIntf) {
  mockData.id = data.id;
  mockData.name = data.name;
  mockData.introduction = data.introduction;
  mockData.modules = data.modules;
}

export const getCurriculum = async (): Promise<DashboardIntf | undefined> => {
  return Promise.resolve(mockData);
};