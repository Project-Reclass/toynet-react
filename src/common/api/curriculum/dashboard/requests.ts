import axios from 'axios';
import { DashboardIntf } from 'src/common/types/curriculum';

export const getCurriculum = async (curriculumId: number): Promise<DashboardIntf | undefined> => {
  const { data } = await axios.get(`/data/curriculum/${curriculumId}.json`);
  return data;
};