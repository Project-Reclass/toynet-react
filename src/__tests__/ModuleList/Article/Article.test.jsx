import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render, waitFor } from '@testing-library/react';

import Article from 'src/Curriculum/Article';
jest.mock('src/common/api/curriculum/article/requests');
import { getArticleText, getArticleMeta } from 'src/common/api/curriculum/article/requests';

const articleId = 42;

const mockArticle = {
  text: 'I have the high ground!',
  meta: {
    source: 'https://obiwan.gov',
    title: 'Return of the Sith',
    author: 'Obi-Wan Kenobi',
    images: [],
  }
}

const RenderWithRouter = ({ children, moduleId, articleId }) => (
  <MemoryRouter initialEntries={[`/module/${moduleId}/article/${articleId}`]}>
    <Route path="/module/:moduleId/article/:articleId">{children}</Route>
  </MemoryRouter>
);

describe('The Article page', () => {
  beforeEach(() => {
    getArticleText.mockClear();
    getArticleMeta.mockClear();

    getArticleMeta.mockResolvedValue(mockArticle.meta);
    getArticleText.mockResolvedValue(mockArticle.text);
  });
  it('should match snapshot', async () => {
    const { getByText, container } = render(
      <RenderWithRouter articleId={articleId}>
        <Article />
      </RenderWithRouter>
    );

    await waitFor(() => expect(getByText(mockArticle.text)).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  })
  it('should get article text and meta for article id specified in the URL', async () => {
    render(
      <RenderWithRouter articleId={articleId}>
        <Article />
      </RenderWithRouter>
    );

    await waitFor(() =>  expect(getArticleMeta.mock.calls[0][0]).toBe(articleId));
    await waitFor(() => expect(getArticleText.mock.calls[0][0]).toBe(articleId));
  });
  it('should render the text of the article', async () => {
    const { getByText } = render(
      <RenderWithRouter articleId={articleId}>
        <Article />
      </RenderWithRouter>
    );

    await waitFor(() => expect(getByText(mockArticle.text)).toBeInTheDocument());
  });
  it('should render the author of the article', async () => {
    const { getByText } = render(
      <RenderWithRouter articleId={articleId}>
        <Article />
      </RenderWithRouter>
    );

    await waitFor(() => expect(getByText(mockArticle.meta.author, { exact: false })).toBeInTheDocument());
  });
  it('should render the title of the article', async () => {
    const { getByText } = render(
      <RenderWithRouter articleId={articleId}>
        <Article />
      </RenderWithRouter>
    );

    await waitFor(() => expect(getByText(mockArticle.meta.title)).toBeInTheDocument());
  });
});