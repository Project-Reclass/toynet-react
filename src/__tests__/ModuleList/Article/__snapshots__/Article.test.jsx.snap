// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Article page should match snapshot 1`] = `
<div>
  <div
    class="css-u6v8er"
  >
    <h1
      class="css-1toumeq"
    >
      Return of the Sith
    </h1>
    <h2
      class="css-1smupne"
    >
      Author: 
      Obi-Wan Kenobi
    </h2>
    <h3
      class="css-tua7mj"
    >
      <p>
        I have the high ground!
      </p>
    </h3>
  </div>
</div>
`;
